sql/pgwire/types.go

```diff
@@ -105,9 +106,11 @@ func (b *writeBuffer) writeDatum(d driver.Datum) error {
 		return b.WriteByte('f')
 
 	case *driver.Datum_IntVal:
-		b.putInt32(8)
-		b.putInt64(v.IntVal)
-		return nil
+		// start at offset 4 because `putInt32` clobbers the first 4 bytes.
+		s := strconv.AppendInt(b.putbuf[4:4], v.IntVal, 10)
```

Comment:
`strconv.AppendInt` implies an allocation and this is inner loop code. Worth a TODO to explore something like `appendUint`. I bet the `Select1_Cockroach` benchmarks show a hit with this change.